llvm/cmake/config-ix.cmake

```diff
@@ -167,7 +167,42 @@ if(LLVM_USE_STATIC_ZSTD AND NOT TARGET zstd::libzstd_static)
   endif()
 set(LLVM_ENABLE_ZSTD OFF)
 else()
-set(LLVM_ENABLE_ZSTD ${zstd_FOUND})
+  # Check if the zstd static library is usable when BUILD_SHARED_LIBS is ON
+  # Test linking zstd::libzstd_static with a shared library. This is to ensure
+  # that the static library is built with -fPIC flag.
```

Comment:
i havent seen this kind of issue before, is that because almost everything is built with `-fPIC` so it just works?

are there any other examples in llvm or other open source projects of solving it this way?

Reply:
> i havent seen this kind of issue before, is that because almost everything is built with `-fPIC` so it just works?
> 
This issue was observed when DPC++ is built in shared lib configuration on Ubuntu 24.04. Apparently, the `zstd` static library that comes with Ubuntu 24.04 was built without the `-fPIC` flag and thus it gives error when linking to dynamic libraries. I've filed a bug report about the zstd package in Ubuntu 24.04 here: https://bugs.launchpad.net/ubuntu/+source/libzstd/+bug/2086543

We did not observe this error before because the `zstd` package that we use in CI (Ubuntu 22.04) was built correctly with `fPIC` flag.

> are there any other examples in llvm or other open source projects of solving it this way?

I'm not aware of any. Let me dig around.

Reply:
yeah my overalll point is i've never seen this kind of issue before so i don't know what the fix should look like so i can't give a good review :)